Sources/MapboxMaps/Foundation/MapProjection.swift

```diff
@@ -6,7 +6,12 @@
 /// Describes the projection used to render the map.
 ///
 /// Mapbox map supports Mercator and Globe projections.
-@_spi(Experimental) public enum MapProjection: Codable, Hashable {
+@_spi(Experimental) public enum MapProjection: Codable, Hashable, CaseIterable {
+    public static var allCases: [MapProjection] = [.mercator(), .globe()]
+    enum CodingKeys: String, CodingKey {
+        case name
+    }
+
```

Comment:
Why are these changes necessary?

Reply:
I see now. I opened a discussion with GL Native team about this. Even if we don't change GL Native for the beta, I'd like to avoid adding the allCases as public API (let it be an implementation detail in `Style.mapProjection()`) since this enum is designed to be able to be extended with additional projections that may contain configuration options.